src/Debugging-Core/Context.extension.st

```diff
@@ -304,16 +309,23 @@ Context >> stepToHome: aContext [
 
 { #category : #'*Debugging-Core' }
 Context >> stepToSendOrReturn [
+
 	"Simulate the execution of bytecodes until either sending a message or 
 	returning a value to the receiver (that is, until switching contexts)."
 
 	| context |
-	[ self isDead or: [ self willSend or: [  self willReturn or: [ self willStore or: [self willCreateBlock ] ] ] ] ]
-		whileFalse: [
+	self method isQuick ifTrue: [ 
+		self method primitive = self returnSelfPrimitive ifFalse: [ 
+			pc := pc + self quickMethodPrimitiveBytecodeSize.
```

Comment:
I wonder if this works for other quick primitives too, like return false, true, accessors... Also I don't remember if very used numbers (0, 1, ...) are compiled as quick too or not.

Reply:
No it doesn't, it is more complicated than anticipated.
Requires more work.

Reply:
There are two cases if I understand it right:
- quick prims that push something on the stack then return it (like an instvar or a number), these methods executes two bytecodes: the push and the return
- quick prims with optimized push+return, that only execute one bytecode: `return:true` or `returnSelf`

For the first case, when we restart a quick method context we need to put the pc at the first bytecode so we have to skip the primitive: `pc := pc + quickPrimitiveSize`.

For the second case, when we restart a quick method context, we just need to go at the end of the method: `pc := pc + self method endPC`

Reply:
I tried to make it simple, and fixed directly related problems in the way.
It can be reviewed already.